localstack-core/localstack/services/cloudformation/engine/v2/change_set_model_executor.py

```diff
@@ -389,22 +388,27 @@ def _execute_resource_action(
                     reason,
                     exc_info=LOG.isEnabledFor(logging.DEBUG),
                 )
-                stack = self._change_set.stack
-                stack.set_resource_status(
-                    logical_resource_id=logical_resource_id,
-                    # TODO,
-                    physical_resource_id="",
-                    resource_type=resource_type,
-                    status=ResourceStatus.CREATE_FAILED
-                    if action == ChangeAction.Add
-                    else ResourceStatus.UPDATE_FAILED,
-                    resource_status_reason=reason,
-                )
```

Comment:
The processing of a failed event is handled at a level above. So this lines just add repetition to that